### PR TITLE
Resnet18 Charge and Time

### PR DESCRIPTION
**Pull request by adelorenzis**
## What?
I have added the code to 1) use time information besides the charge; 2) collapse all the 19 channels of charge and/or time into 2 channels each (made of mean and standard deviation);  3) rescale features in a standardized form.

## Why?
Addition has been made to include full information: charge and time for each PMT. Using all data is found to increase the accuracy of the ResNet. Results were reported in WatchMal meetings.

## How?

I have introduced four new options in class CNNmPMTDataset: "mode", "collapse_mode", "scaling_charge", "scaling_time"

"mode" can be ['charge', 'time'], ['charge'], ['time']
 
"collapse_mode" can be [], ['charge', 'time'], ['charge'], ['time']

"scaling_charge" can be [], [mean_value, std_value]

"scaling_time" can be [], [mean_value, std_value]

The previous option "collapse_array" is not used anymore. It is left in the code for possible compatibility issues.

The main changes affect the method "process_data".

In particular, according to the content of "mode", I include the charge, the time or both, and eventually I concatenate them together.
According to the content of "collapse_mode", the corresponding channels are rescaled through standardization. For this purpose, the function feature_scaling_std has been introduced. It takes as input mean and std extracted from "scaling_charge" and/or "scaling_time".

The use of "apply_random_transformations" has been replaced with "apply_random_transformations_fixedchoices" to make sure that the same transformations are applied to both charge and time data.
The function "apply_random_transformations_fixedchoices" is defined in the file data_utils.py

### CODE REVIEW

**Comment by pdeperio** [comment]
Thanks @adelorenzis @mpcasado! 

Please take a look at e.g. [How to Write a Pull Request](https://www.pullrequest.com/blog/writing-a-great-pull-request-description) or [The Anatomy of a PR](https://www.pullrequest.com/blog/writing-a-great-pull-request-description) to get an idea of how to write a description above, including results of testing.

I noticed you also have several `_old` files, which we don't want to store in the repository since I guess we'll never use them. Please just modify the corresponding file directly and commit; that's what this version control is for!

You'll also notice on the PR webpage several "Conflicting files". Please review these carefully to ensure you're putting in the right changes.

**Comment by adelorenzis** [comment]
Hello @pdeperio @mpcasado! 
Sorry for the delay in response.
Thanks for your comments and suggestions.
I did a new pull request, I added a description, I removed the old files and I reviewed the "conflicting files".
Please, let me know if anything else is needed.

**Comment by pdeperio** [comment]
@adelorenzis Thank you! Can you please resolve the conflicts below?

**Review by nickwp — CHANGES_REQUESTED** [review]

File: watchmal/dataset/cnn_mpmt/cnn_mpmt_dataset.py [hunk 1]
```diff
@@ -72,22 +86,60 @@ def process_data(self, hit_pmts, hit_data):
...
 
     def  __getitem__(self, item):
 
         data_dict = super().__getitem__(item)
 
-        processed_data = from_numpy(self.process_data(self.event_hit_pmts, self.event_hit_charges))
+        # select random choices for transformations for charge and time
+        rand_choices = []
+        if self.transforms is not None:
+            rand_choices = [bool(random.getrandbits(1)) for i in range(len(self.transforms))]
```
[the remark this answers is not in the document]
Comment (adelorenzis) [reply watchmal/dataset/cnn_mpmt/cnn_mpmt_dataset.py]: @nickwp I proposed a new solution and I updated the code with your suggestions. Thanks! Please let me know if it's ok now.

**Comment by nickwp** [comment]
Closing this as it's replaced by #67

**Pull request closed without merging**